climada/util/lines_polys_handler.py

```diff
@@ -0,0 +1,891 @@
+
+    Returns
+    -------
+    exp_pnt : Exposures
+        Exposures with a double index geodataframe, first for the geometries of exp,
+        second for the point disaggregation of the geometries.
+
+    """
+
+    # rasterize
+    if to_meters:
+        gdf_pnt = line_to_pnts_m(gdf, res)
+    else:
+        gdf_pnt = line_to_pnts(gdf, res)
+
+    # disaggregate
+    if disagg == 'avg':
+        gdf_pnt = disagg_gdf_avg(gdf_pnt)
+    elif disagg == 'surf':
+        gdf_pnt = disagg_gdf_val(gdf_pnt, res)
+    elif disagg is None and 'value' not in gdf_pnt.columns:
+        gdf_pnt['value'] = 1
+
+    return gdf_pnt
+
+
+def _gdf_poly_to_pnt(gdf, res, to_meters, disagg):
+    """
+    Disaggregate exposures with (multi-)polygons
+    geometries to points.
+
+    Parameters
+    ----------
+    exp : Exposures
+        The exposure instance with exp.gdf.geometry containing (multi-)polygons
+    res : float
+        Resolution of the disaggregation grid.
+    to_meters : bool
+       If True, the geometries are projected to an equal area projection before
+       the disaggregation. res is then in meters. The exposures are
+       then reprojected into the original projections before the impact
+       calculation.
+    disagg : string
+        Disaggregation method.
+        if 'avg' value is average over points
+        if 'surf' value is surface per points. Area per point (res*res)
+            for polygons.
+        if 'None' value is unchanged (or 1 if no value is defined)
+
+    Returns
+    -------
+    exp_pnt : Exposures
+        Exposures with a double index geodataframe, first for the geometries of exp,
+        second for the point disaggregation of the geometries.
+
+    """
+
+    # rasterize
+    if to_meters:
+        gdf_pnt = poly_to_pnts_m(gdf, res)
+    else:
+        gdf_pnt = poly_to_pnts(gdf, res)
+
+    # disaggregate
+    if disagg == 'avg':
+        gdf_pnt = disagg_gdf_avg(gdf_pnt)
+    elif disagg == 'surf':
+        gdf_pnt = disagg_gdf_val(gdf_pnt, res * res)
+    elif disagg is None and 'value' not in gdf_pnt.columns:
+        gdf_pnt['value'] = 1
+
+    return gdf_pnt
+
+
+def disagg_gdf_avg(gdf_pnts):
+    """
+    Disaggragate value of geodataframes from geoemtries to points
+
+    Parameters
+    ----------
+    gdf_pnts : geodataframe
+        Geodataframe with a double index, first for geometries (lines, polygons),
+        second for the point disaggregation of the polygons. The value column is assumed
+        to represent values per polygon (first index).
+
+    Returns
+    -------
+    gdf_agg : geodataframe
+        The value per geometry are evenly distributed over the points per geometry.
+
+    """
+
+    gdf_agg = gdf_pnts.copy()
+
+    group = gdf_pnts.groupby(axis=0, level=0)
+    gdf = group.value.mean() / group.value.count()
+
+    gdf = gdf.reindex(gdf_pnts.index, level=0)
+    gdf_agg['value'] = gdf
+
+    return gdf_agg
+
+
+def disagg_gdf_val(gdf_pnts, value_per_pnt):
```

Comment:
dito.
E.g. `disagg_vals_cnst`

Reply:
I do not like `cnst` so much as it is not clear what `constant` means imho.

Reply:
`disagg_gdf_values_fixed`?